oxidation/crates/parsec_client_types/src/local_manifest.rs

```diff
@@ -0,0 +1,347 @@
+    fn from(obj: LocalFolderManifest) -> Self {
+        Self {
+            type_: Default::default(),
+            base: obj.base,
+            need_sync: obj.need_sync,
+            updated: obj.updated,
+            children: obj.children,
+            local_confinement_points: Some(obj.local_confinement_points),
+            remote_confinement_points: Some(obj.remote_confinement_points),
+        }
+    }
+}
+
+impl LoadDump for LocalFolderManifest {}
+
+/*
+ * LocalWorkspaceManifest
+ */
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(
+    into = "LocalWorkspaceManifestData",
+    from = "LocalWorkspaceManifestData"
+)]
+pub struct LocalWorkspaceManifest {
+    pub base: WorkspaceManifest,
+    pub need_sync: bool,
+    pub updated: DateTime<Utc>,
+    pub children: HashMap<EntryName, ManifestEntry>,
+    // Confined entries are entries that are meant to stay locally and not be added
+    // to the uploaded remote manifest when synchronizing. The criteria for being
+    // confined is to have a filename that matched the "prevent sync" pattern at the time of
+    // the last change (or when a new filter was successfully applied)
+    pub local_confinement_points: HashSet<ManifestEntry>,
+    // Filtered entries are entries present in the base manifest that are not exposed
+    // locally. We keep track of them to remember that those entries have not been
+    // deleted locally and hence should be restored when crafting the remote manifest
+    // to upload.
+    pub remote_confinement_points: HashSet<ManifestEntry>,
+    // Speculative placeholders are created when we want to access a workspace
+    // but didn't retrieve manifest data from backend yet. This implies:
+    // - non-placeholders cannot be speculative
+    // - the only non-speculative placeholder is the placeholder initialized
+    //   during the initial workspace creation
+    // This speculative information is useful during merge to understand if
+    // a data is not present in the placeholder compared with a remote because:
+    // a) the data is not locally known (speculative is True)
+    // b) the data is known, but has been locally removed (speculative is False)
+    // Prevented to be `required=True` by backward compatibility
+    pub speculative: bool,
+}
+
+new_data_struct_type!(
+    LocalWorkspaceManifestData,
+    type: "local_workspace_manifest",
+    base: WorkspaceManifest,
+    need_sync: bool,
+    #[serde_as(as = "DateTimeExtFormat")]
+    updated: DateTime<Utc>,
+    children: HashMap<EntryName, ManifestEntry>,
+    #[serde(
+        default,
+        deserialize_with = "maybe_field::deserialize_some",
+        serialize_with = "maybe_field::serialize_some"
+    )]
+    local_confinement_points: Option<HashSet<ManifestEntry>>,
+    #[serde(
+        default,
+        deserialize_with = "maybe_field::deserialize_some",
+        serialize_with = "maybe_field::serialize_some"
```

Comment:
no need for the `serialize_with` part given the serialized data will always have those fields (FYI the `LocalDeviceData` also incorrectly have `serialize_with` defined, we should correct this in another PR)

(as a matter of fact we should never need `serialize_with = "maybe_field::serialize_some"` in any schema given we always serialize `None` as `<field_name>: None`)